crates/quantum_info/src/clifford.rs

```diff
@@ -0,0 +1,318 @@
+// This code is part of Qiskit.
+//
+// (C) Copyright IBM 2024
+//
+// This code is licensed under the Apache License, Version 2.0. You may
+// obtain a copy of this license in the LICENSE.txt file in the root directory
+// of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Any modifications or derivative works of this code must retain this
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+use std::fmt;
+
+use ndarray::{azip, s, Array2};
+
+/// Symplectic matrix.
+/// Currently this class is internal to the synthesis library.
+pub struct SymplecticMatrix {
+    /// Number of qubits.
+    pub num_qubits: usize,
+    /// Matrix with dimensions (2 * num_qubits) x (2 * num_qubits).
+    pub smat: Array2<bool>,
+}
+
+/// Clifford.
+/// Currently this class has a very different functionality from Qiskit's
```

Comment:
I would say something like "reduced functionality" and not "very different"

Reply:
done